test/integration/integration_test.cc

```diff
@@ -521,6 +521,14 @@ TEST_P(IntegrationTest, TestSmuggling) {
     sendRawHttpAndWaitForResponse(lookupPort("http"), full_request.c_str(), &response, false);
     EXPECT_THAT(response, StartsWith("HTTP/1.1 400 Bad Request\r\n"));
   }
+  {
+    std::string response;
+    const std::string full_request = "GET / HTTP/1.1\r\nHost: host\r\ncontent-length: "
```

Comment:
I think that in order to smuggle a request you need either content-length: 0 and possibly a request body that is chunk encoded prior to the smuggled request.  A 400 here could be due to the request not being chunk encoded despite there being a transfer-encoding: chunked header.